File: books/javascript/src/examples/js/pop_push.js
"use strict";

var names = [ "foo" ];
console.log(names);   // [ "foo" ]

names[1] = "bar";
console.log(names);   // [ "foo", "bar" ]

names.push("moo");
console.log(names);  // [ "foo", "bar", "moo" ]

names.push("qux", "zorg");
console.log(names); // [ "foo", "bar", "moo", "qux", "zorg" ]

var last = names.pop();
console.log(last);  // "zorg"
console.log(names); // [ "foo", "bar", "moo", "qux" ]

File: books/perl/src/examples/perlarrays/pop_push.pl
#!/usr/bin/perl
use strict;
use warnings;

my @names = ("Foo", "Bar", "Baz");

my $last_name = pop @names;

print "$last_name\n";        # Baz
print "@names\n";            # Foo Bar


push @names, "Moo";

print "@names\n";            # Foo Bar Moo
